compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/propagate_data_layout.mlir

```diff
@@ -5,36 +5,86 @@
+    %extracted_slice = tensor.extract_slice %arg0[%arg3, 0] [64, 512] [1, 1] : tensor<1024x512xi8> to tensor<64x512xi8>
+    %extracted_slice_0 = tensor.extract_slice %arg1[0, %arg4] [512, 64] [1, 1] : tensor<512x1024xi8> to tensor<512x64xi8>
+    %extracted_slice_1 = tensor.extract_slice %0[%arg3, %arg4] [64, 64] [1, 1] : tensor<1024x1024xi32> to tensor<64x64xi32>
+    %2 = linalg.fill ins(%c0_i32 : i32) outs(%extracted_slice_1 : tensor<64x64xi32>) -> tensor<64x64xi32>
+    %3 = tensor.empty() : tensor<1x16x64x32xi8>
+    %pack = tensor.pack %extracted_slice inner_dims_pos = [0, 1] inner_tiles = [64, 32] into %3 : tensor<64x512xi8> -> tensor<1x16x64x32xi8>
+    %4 = tensor.empty() : tensor<16x1x64x32xi8>
+    %5 = tensor.empty() : tensor<16x1x32x64xi8>
+    %pack_2 = tensor.pack %extracted_slice_0 outer_dims_perm = [0, 1] inner_dims_pos = [0, 1] inner_tiles = [32, 64] into %5 : tensor<512x64xi8> -> tensor<16x1x32x64xi8>
+    %6 = tensor.empty() : tensor<1x1x64x64xi32>
+    %pack_3 = tensor.pack %2 inner_dims_pos = [0, 1] inner_tiles = [64, 64] into %6 : tensor<64x64xi32> -> tensor<1x1x64x64xi32>
+    %7 = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d2, d3, d5)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d2, d1, d5, d4)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d3, d4)>], iterator_types = ["parallel", "parallel", "reduction", "parallel", "parallel", "reduction"]} ins(%pack, %pack_2 : tensor<1x16x64x32xi8>, tensor<16x1x32x64xi8>) outs(%pack_3 : tensor<1x1x64x64xi32>) {
+    ^bb0(%in: i8, %in_6: i8, %out: i32):
+      %9 = arith.extsi %in : i8 to i32
+      %10 = arith.extsi %in_6 : i8 to i32
+      %11 = arith.muli %9, %10 : i32
+      %12 = arith.addi %out, %11 : i32
+      linalg.yield %12 : i32
+    } -> tensor<1x1x64x64xi32>
+    %unpack = tensor.unpack %7 inner_dims_pos = [0, 1] inner_tiles = [64, 64] into %2 : tensor<1x1x64x64xi32> -> tensor<64x64xi32>
+    %extracted_slice_4 = tensor.extract_slice %arg2[%arg3, %arg4] [64, 64] [1, 1] : tensor<1024x1024xi32> to tensor<64x64xi32>
+    %extracted_slice_5 = tensor.extract_slice %arg5[%arg3, %arg4] [64, 64] [1, 1] : tensor<1024x1024xi32> to tensor<64x64xi32>
+    %8 = linalg.generic {indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>, affine_map<(d0, d1) -> (d0, d1)>, affine_map<(d0, d1) -> (d0, d1)>], iterator_types = ["parallel", "parallel"]} ins(%unpack, %extracted_slice_4 : tensor<64x64xi32>, tensor<64x64xi32>) outs(%extracted_slice_5 : tensor<64x64xi32>) {
+    ^bb0(%in: i32, %in_6: i32, %out: i32):
+      %9 = arith.addi %in, %in_6 : i32
+      linalg.yield %9 : i32
+    } -> tensor<64x64xi32>
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %8 into %arg5[%arg3, %arg4] [64, 64] [1, 1] : tensor<64x64xi32> into tensor<1024x1024xi32>
+    }
+  } {mapping = [#gpu.block<y>, #gpu.block<x>]}
+  return %1 : tensor<1024x1024xi32>
+}
+
+// CHECK-LABEL: matmul_elementwise_1024x1024x512_i8xi8xi32
```

Comment:
I know the existing tests dont do that but it would be better to have better test here... Basically capture the results of the first generic and make sure it is used in the second generic op.

Reply:
Done.